mavutil.py

```diff
@@ -92,6 +92,9 @@ def __str__(self):
 
 def add_message(messages, mtype, msg):
     '''add a msg to array of messages, taking account of instance messages'''
+    if msg.get_type() == 'BAD_DATA':
+        # avoid dealing with instance field
```

Comment:
Have you made sure that `BAD_DATA` still appears in MAVProxy's `messages` command after this change?

Reply:
Hi @peterbarker -- I added the stack trace and some notes to the linked issue.

Yeah, I didn't test the "bad comms" case. I will need to rig something up to test this.

I need to spend more time thinking about the appropriate fix. Feel free to close this PR.

Happy New Year!